From 7f462cf36b9529cb2aec4984e2fb78b8216dfe69 Mon Sep 17 00:00:00 2001
From: David S. Wang <dsw@cloudera.com>
Date: Wed, 27 Mar 2013 09:33:56 -0700
Subject: [PATCH 19/96] HBASE-7152 testShouldCheckMasterFailOverWhenMETAIsInOpenedState times out occasionally

Reason: test
Author: Jimmy Xiang
Ref: CDH-8819
(cherry picked from commit bd3a21d8e191a33efc2ee3c3683ca4ecb96d50bb)
---
 .../hadoop/hbase/master/TestMasterFailover.java    |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/test/java/org/apache/hadoop/hbase/master/TestMasterFailover.java b/src/test/java/org/apache/hadoop/hbase/master/TestMasterFailover.java
index 479d904..91577cd 100644
--- a/src/test/java/org/apache/hadoop/hbase/master/TestMasterFailover.java
+++ b/src/test/java/org/apache/hadoop/hbase/master/TestMasterFailover.java
@@ -108,6 +108,10 @@ public class TestMasterFailover {
     LOG.info("Staring cluster for second time");
     TEST_UTIL.startMiniHBaseCluster(NUM_MASTERS, NUM_RS);
 
+    HMaster master = TEST_UTIL.getHBaseCluster().getMaster();
+    while (!master.isInitialized()) {
+      Thread.sleep(100);
+    }
     // Failover should be completed, now wait for no RIT
     log("Waiting for no more RIT");
     ZKAssign.blockUntilNoRIT(zkw);
-- 
1.7.0.4

